aten/src/ATen/native/cuda/LogAddExpKernel.cu

```diff
@@ -35,13 +36,14 @@ void logaddexp2_kernel_cuda(TensorIteratorBase& iter) {
       iter.dtype(), "logaddexp2_cuda",
       [&]() {
         using accscalar_t = at::acc_type<scalar_t, /*is_cuda=*/true>;
-        gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a, scalar_t b) -> scalar_t {
+        const auto inv_log_2 = static_cast<accscalar_t>(1.0 / c10::ln_2<double>);
+        gpu_kernel(iter, [inv_log_2] GPU_LAMBDA (scalar_t a, scalar_t b) -> scalar_t {
           if (::isinf(static_cast<accscalar_t>(a)) && a == b) {
             return a;
           }
           else {
             scalar_t m = ::max(a, b);
-            return m + ::log2((scalar_t)(1.0) + ::pow((scalar_t)(2.0), -::abs(a - b)));
+            return m + ::log1p(::exp2(-::abs(a - b))) * inv_log_2;
```

Comment:
it's a pre-existing condition, but looks like intermediate computations are not done in fp32, can you fix it here and in logaddexp (also replace accscalar with opmath)?